.github/workflows/pr.yaml

```diff
@@ -0,0 +1,93 @@
+name: PR checks
+on: pull_request
+env:
+  SECRET_KEY: insecure_dev_key
+
+jobs:
+  run-image:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build image
+        run: DOCKER_BUILDKIT=1 docker build --tag anbox-cloud-io .
+
+      - name: Run image
+        run: |
+          docker run --detach --env SECRET_KEY=insecure_dev_key --network host anbox-cloud-io
+          sleep 1
+          curl --head --fail --retry-delay 1 --retry 30 --retry-connrefused http://localhost
+
+  run-dotrun:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install dotrun
+        run: |
+          sudo snap install dotrun
+          sudo chown root:root /
+
+      - name: Install dependencies
+        run: dotrun install
+
+      - name: Build assets
+        run: dotrun build
+
+      - name: Test site
+        run: dotrun & curl --head --fail --retry-delay 1 --retry 30 --retry-connrefused http://localhost:8010
+
+  lint-scss:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      - name: Lint scss
+        run: yarn lint-scss
+
+  lint-python:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install node dependencies
+        run: yarn install --immutable
+
+      - name: Install python dependencies
+        run: sudo pip3 install flake8 black
+
+      - name: Lint python
+        run: yarn lint-python
+
+  test-python:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install requirements
+        run: |
+          sudo apt-get update && sudo apt-get install --yes python3-setuptools
+          sudo pip3 install -r requirements.txt
+
+      - name: Install dependencies
+        run: sudo pip3 install coverage pyyaml==5.3.1
```

Comment:
@nottrobin this is the only way I could make it work, pinning the version of pyyaml, as you mentioned yesterday. It only happens with `test-python`

Reply:
We really should change requirements.txt if the app actually relies on pyyaml, which I believe it does.